When QNNPACK is not available the test will fail trying to use it.
Skip them conditionally.
See https://github.com/pytorch/pytorch/pull/82882

Author: Alexander Grund (TU Dresden)

diff --git a/test/test_nnapi.py b/test/test_nnapi.py
index f8db7e1a3d..1d4fd52e28 100644
--- a/test/test_nnapi.py
+++ b/test/test_nnapi.py
@@ -2,8 +2,10 @@
 import os
 import ctypes
 import torch
+import unittest
 from typing import Tuple
 from torch.backends._nnapi.prepare import convert_model_to_nnapi
+from torch.testing._internal.common_quantized import supported_qengines
 from torch.testing._internal.common_utils import TestCase, run_tests
 
 
@@ -18,6 +20,8 @@ def nhwc(t):
     return t
 
 
+@unittest.skipUnless('qnnpack' in supported_qengines,
+                     "This Pytorch Build has not been built with or does not support QNNPACK")
 class TestNNAPI(TestCase):
 
     def setUp(self):
